services/repository/src/main/java/org/sagebionetworks/auth/filter/AcceptTermsOfUseFilter.java

```diff
@@ -55,7 +55,10 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 		// If the user is not anonymous, check if they have accepted the terms of use
 		if (!BOOTSTRAP_PRINCIPAL.ANONYMOUS_USER.getPrincipalId().equals(userId)) {
 			if (!authenticationService.hasUserAcceptedTermsOfService(userId)) {
-				HttpAuthUtil.rejectWithErrorResponse(httpResponse, TOU_UNSIGNED_REASON, HttpStatus.FORBIDDEN);
+				String hostName = httpRequest.getServerName();
+				HttpAuthUtil.rejectWithErrorResponse(httpResponse,
+						String.format(TOU_UNSIGNED_REASON, HttpAuthUtil.synapseHostFromRepoHostName(hostName)),

```

Comment:
Having second thoughts... Unnecessary complicated, could just put generic ' Login to Synapse' and be done with it.